# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/threads/SystemGroup.cpp
# Commit: 36b0646e8649
# Full Hash: 36b0646e86498662e770756ad54b1fd648a82184
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-04-07 21:44:02
# Regressor Bug: 1620594
# File Overlap Count: 1
# Description:
#   Bug 1620594 - Part 7: Remove TabGroup and SystemGroup. r=nika,bas
#   
#   TabGroup never really made any difference in which thread something go
#   dispatched to. This was the intended use, but development of TabGroups
#   with abstract main threads never made it that far. The good thing is
# ==============================================================================

diff -r a5ad3b179c92 -r 36b0646e8649 xpcom/threads/SystemGroup.cpp
--- a/xpcom/threads/SystemGroup.cpp	Tue Apr 07 15:17:14 2020 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,85 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set ts=8 sts=2 et sw=2 tw=80: */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#include "SystemGroup.h"
-
-#include <utility>
-
-#include "mozilla/AbstractThread.h"
-#include "mozilla/StaticPtr.h"
-#include "mozilla/UniquePtr.h"
-
-using namespace mozilla;
-
-class SystemGroupImpl final : public SchedulerGroup {
- public:
-  SystemGroupImpl();
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(SystemGroupImpl, override)
-
-  static void InitStatic();
-  static void ShutdownStatic();
-  static SystemGroupImpl* Get();
-
-  static bool Initialized() { return !!sSingleton; }
-
- private:
-  ~SystemGroupImpl() = default;
-  static StaticRefPtr<SystemGroupImpl> sSingleton;
-};
-
-StaticRefPtr<SystemGroupImpl> SystemGroupImpl::sSingleton;
-
-SystemGroupImpl::SystemGroupImpl() {
-  CreateEventTargets(/* aNeedValidation = */ true);
-}
-
-/* static */
-void SystemGroupImpl::InitStatic() {
-  MOZ_ASSERT(!sSingleton);
-  MOZ_ASSERT(NS_IsMainThread());
-  sSingleton = new SystemGroupImpl();
-}
-
-/* static */
-void SystemGroupImpl::ShutdownStatic() {
-  sSingleton->Shutdown(true);
-  sSingleton = nullptr;
-}
-
-/* static */
-SystemGroupImpl* SystemGroupImpl::Get() {
-  MOZ_ASSERT(sSingleton);
-  return sSingleton.get();
-}
-
-void SystemGroup::InitStatic() { SystemGroupImpl::InitStatic(); }
-
-void SystemGroup::Shutdown() { SystemGroupImpl::ShutdownStatic(); }
-
-bool SystemGroup::Initialized() { return SystemGroupImpl::Initialized(); }
-
-/* static */
-nsresult SystemGroup::Dispatch(TaskCategory aCategory,
-                               already_AddRefed<nsIRunnable>&& aRunnable) {
-  if (!SystemGroupImpl::Initialized()) {
-    return NS_DispatchToMainThread(std::move(aRunnable));
-  }
-  return SystemGroupImpl::Get()->Dispatch(aCategory, std::move(aRunnable));
-}
-
-/* static */
-nsISerialEventTarget* SystemGroup::EventTargetFor(TaskCategory aCategory) {
-  if (!SystemGroupImpl::Initialized()) {
-    return GetMainThreadSerialEventTarget();
-  }
-  return SystemGroupImpl::Get()->EventTargetFor(aCategory);
-}
-
-/* static */
-AbstractThread* SystemGroup::AbstractMainThreadFor(TaskCategory aCategory) {
-  MOZ_ASSERT(SystemGroupImpl::Initialized());
-  return SystemGroupImpl::Get()->AbstractMainThreadFor(aCategory);
-}